src/OrchardCore.Modules/OrchardCore.Contents/Liquid/BuildDisplayFilter.cs

```diff
@@ -45,9 +45,19 @@ static async ValueTask<FluidValue> Awaited(Task<IShape> task)
                 throw new ArgumentException("Services missing while invoking 'shape_build_display'");
             }
 
+            var serviceProvider = (IServiceProvider)services;
+
+            var buildDisplayRecursionHelper = serviceProvider.GetRequiredService<IContentItemRecursionHelper<BuildDisplayFilter>>();
+
+            // When {{ Model.ContentItem | shape_build_display | shape_render }} is called prevent recursion.
+            if (buildDisplayRecursionHelper.IsRecursive(contentItem))
+            {
+                return new ValueTask<FluidValue>(NilValue.Instance);
```

Comment:
Are we sure here that it can't be called more than once, not necessarily through the same model / shape, but on the same item, this in a given request scope?